src/AppBundle/Entity/Repository/DocumentRepository.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace AppBundle\Entity\Repository;
+
+use AppBundle\Entity\Report\Document;
+use Doctrine\ORM\EntityRepository;
+
+class DocumentRepository extends EntityRepository
+{
+    /**
+     * Get soft-deleted documents
+     *
+     * @return Document[]
+     */
+    public function retrieveSoftDeleted()
+    {
+        $qb = $this->createQueryBuilder('d')
+                ->where('d.deletedAt IS NOT NULL');
+
+        $filter = $this->_em->getFilters()->enable('softdeleteable');
+        $filter->disableForEntity(Document::class);
+        $records = $qb->getQuery()->getResult(); /* @var $records Document[] */
+        $this->_em->getFilters()->enable('softdeleteable');
+
+        return $records;
+    }
+
+    /**
+     * @param $id document ID
+     */
+    public function hardDeleteDocument($id)
+    {
+        $this->_em->clear();
+        $filter = $this->_em->getFilters()->enable('softdeleteable');
```

Comment:
Yes, weird. I followed the example here
http://atlantic18.github.io/DoctrineExtensions/doc/softdeleteable.html
I'll play with alternatives